llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -6005,10 +6005,12 @@ void BoUpSLP::reorderTopToBottom() {
       if ((TE->State == TreeEntry::Vectorize ||
            TE->State == TreeEntry::StridedVectorize) &&
           isa<ExtractElementInst, ExtractValueInst, LoadInst, StoreInst,
-              InsertElementInst>(TE->getMainOp())) {
+              InsertElementInst, ShuffleVectorInst>(TE->getMainOp())) {
         assert(!TE->isAltShuffle() &&
                "Alternate instructions are only supported by BinaryOperator "
                "and CastInst.");
+        assert(!isa<ShuffleVectorInst>(TE->getMainOp()) ||
+               SLPReVec && "Only supported by REVEC.");
```

Comment:
It should be not an assertion, but a logic in the if condition

Reply:
I don't get it. Only REVEC can support `ShuffleVectorInst`, why do not use assert here?

Reply:
Without revec the compiler should not crash on shuffles, it just shall ignore them

Reply:
How does the TreeEntry has `ShuffleVectorInst` as Scalars without REVEC? `ShuffleVectorInst` returns vector.

Reply:
It may help for constant shuffles in some cases, I assume